src/napari_matplotlib/tests/test_theme.py

```diff
@@ -88,3 +90,31 @@ def test_titles_respect_theme(
 
     assert ax.xaxis.label.get_color() == expected_text_colour
     assert ax.yaxis.label.get_color() == expected_text_colour
+
+
+@pytest.mark.mpl_image_compare
+def test_no_theme_side_effects(make_napari_viewer):
+    """Ensure that napari-matplotlib doesn't pollute the globally set style.
+
+    A MWE to guard aganst issue matplotlib/#64. Should always reproduce a plot
+    with the default matplotlib style.
+    """
+    import matplotlib.pyplot as plt
+
+    np.random.seed(12345)
+
+    # should not affect global matplotlib plot style
+    viewer = make_napari_viewer()
+    viewer.theme = "dark"
+    NapariMPLWidget(viewer)
+
+    # some plotting unrelated to napari-matplotlib
+    image = np.random.random((3, 3))
+    plt.imshow(image)
```

Comment:
It's a bit safer in terms of knowing what figure you're referring to later to explicitly create a new figure here using `fig, ax = plt.subplots()`, then use `ax.various_methods(...)` to create the plot and pass `fig` to deepcopy()` without the `gcf()` call. You might not even need the `deepcopy` in this case because the figure isn't related to a napari widget.